input/cql/IMMZDT14.cql

```diff
@@ -105,37 +94,56 @@ define "2 doses":
  * References:
  * 	- WHO recommendations for routine immunization - summary tables: https://w ww.who.int/teams/immunization-vaccines-and-biologicals/policies/who-recommendations-for-routine-immunization---summary-tables
  * Logic:
- *	 ((((("Typhoid vaccine immunization history" = "No-doses") and ("Age" > "6 years")) and ("Pregnant" = "False")) or (((("Typhoid vaccine immunization history" = "1 dose") and ("Previous dose date" = "2 days")) and ("Age" > "6 years")) and ("Pregnant" = "False"))) or (((("Typhoid vaccine immunization history" = "2 doses" = true) and ("Previous dose date" = "2 days")) and ("Age" > "6 years")) and ("Pregnant" = "False")))
- */
-define "Should vaccinate patient with Ty21a vaccine":
-	((((("Typhoid vaccine immunization history" = "No-doses") and ("Age" > "6 years")) and ("Pregnant" = "False")) or (((("Typhoid vaccine immunization history" = "1 dose") and ("Previous dose date" = "2 days")) and ("Age" > "6 years")) and ("Pregnant" = "False"))) or (((("Typhoid vaccine immunization history" = "2 doses" = true) and ("Previous dose date" = "2 days")) and ("Age" > "6 years")) and ("Pregnant" = "False")))
-
-/* 
- * @dataElement No doses
- */
-define "No doses":
-	0 // TODO: Define this
-
-define "6 months":
-	0
-
-/* 
- * @dataElement 45 years
+ *	 ((((("Typhoid vaccine immunization history" = "No-doses") and ("Current Patient Age in Years" > 6)) and ("Pregnancy Status" = false)) or ((((("Typhoid vaccine immunization history" = "1 dose") and ("Date last Typhoid dose given" > "1 day")) and ("Date last Typhoid dose given" < "21 days")) and ("Type of last Typhoid dose" = "Ty21a")) and ("Pregnancy Status" = false))) or ((((("Typhoid vaccine immunization history" = "2 doses" = true) and ("Date last Typhoid dose given" > "1 day")) and ("Date last Typhoid dose given" < "21 days")) and ("Type of last Typhoid dose" = "Ty21a")) and ("Pregnancy Status" = false)))
+ */
+define "Should vaccinate patient with Ty21a vaccine on a 3 dose scheme":
```

Comment:
@mo3dev / @nityan  - It appears that this logic can be reduced to handle the cases where there are no doses, or 1 or 2 doses - I have tried to extract out the common clauses - are they still accurate?